groupcache/src/service_discovery.rs

```diff
@@ -0,0 +1,10 @@
+use crate::GroupcachePeer;
+use async_trait::async_trait;
+
+#[async_trait]
+pub trait ServiceDiscovery {
```

Comment:
This would need documentation along - would be nice to provide example in code above the definition returning statis ips

Reply:
+ tests similar as for other public api :)